### PR TITLE
ENG-3792 change mfe routing path assembling

### PR DESCRIPTION
**Pull request by ichalagashvili**

### CODE REVIEW

File: src/helpers/urlUtils.js [hunk 1]
```diff
@@ -1,3 +1,14 @@
 import { get } from 'lodash';
-// eslint-disable-next-line import/prefer-default-export
+
 export const adminConsoleUrl = url => `${get(process.env, 'DOMAIN', '')}/${url}`;
+
+export const generateMfeRoutes = mfe => mfe.reduce((acc, curr) => {
+  if (curr.paths) {
+    const routes = [];
+    curr.paths.forEach((route) => {
+      routes.push({ route: `/${curr.pbcName}/${curr.bundleName}${route}`, id: curr.id || `id-${curr.pbcName}-${curr.bundleName} - ${route}` });
```
Comment (Dv92) [inline src/helpers/urlUtils.js]: sometimes `route` can start with `/` and other times could be without, this depends on how the user have written the path in the bundle descriptor. So we must prevent issues in the interpolation, for this reason I suggest you to clean the route and create the url as a `join` in this way:
```javascript
const path = route.split("/").filter(Boolean).join("/")
const route = [curr.pbcName, curr.bundleName, path].join("/")

routes.push({ route: `/${route}`, id: curr.id ||`id-${curr.pbcName}-${curr.bundleName}-${route}` });
```
you can do the same thing for the `id` but with "-", also remove the spaces inside the `id`

Comment (ichalagashvili) [reply src/helpers/urlUtils.js]: done ✅

**Review by Dv92 — DISMISSED** [review]

**Comment by Dv92** [comment]
@ichalagashvili please, can you see the coverage issue?

File: src/ui/app/App.js [hunk 2]
```diff
@@ -205,17 +206,12 @@ const RouteComponent = () => {
...
-        routes.push({ route, id: curr.id });
-      });
-      return [...acc, ...routes];
-    }
-    return acc;
+  const mfeRoutes = React.useMemo(
+    () => generateMfeRoutes(contentMfe),
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []), [JSON.stringify(contentMfe)]);
+    [],
```
Comment (gReis89) [inline src/ui/app/App.js]: should it be removed? looks like you are passing 3 parameters to `useMemo` instead of 2.

File: src/ui/app/App.js [hunk 3]
```diff
@@ -205,17 +206,10 @@ const RouteComponent = () => {
...
-        routes.push({ route, id: curr.id });
-      });
-      return [...acc, ...routes];
-    }
-    return acc;
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []), [JSON.stringify(contentMfe)]);
+  const mfeRoutes = React.useMemo(
+    () => generateMfeRoutes(contentMfe),
+    [contentMfe],
```
Comment (gReis89) [inline src/ui/app/App.js]: this is an array of object, I think it should be stringified in order to correctly memoise it, what do u think @Dv92 ?

**Review by gReis89 — DISMISSED** [review]
looks good!

**Review by gReis89 — DISMISSED** [review]
LGTM

**Review by gReis89 — APPROVED** [review]

**Pull request merged**